pilot/pkg/networking/core/v1alpha3/waypoint.go

```diff
@@ -38,27 +40,40 @@ const (
 	ConnectUpgradeType = "CONNECT"
 )
 
+type waypointServices struct {
+	services        map[host.Name]*model.Service
+	orderedServices []*model.Service
+}
+
 // findWaypointResources returns workloads and services associated with the waypoint proxy
-func findWaypointResources(node *model.Proxy, push *model.PushContext) ([]*model.WorkloadInfo, map[host.Name]*model.Service) {
+func findWaypointResources(node *model.Proxy, push *model.PushContext) ([]*model.WorkloadInfo, *waypointServices) {
 	scope := node.WaypointScope()
 	workloads := push.WorkloadsForWaypoint(scope)
 	return workloads, findWorkloadServices(workloads, push)
 }
 
-func findWorkloadServices(workloads []*model.WorkloadInfo, push *model.PushContext) map[host.Name]*model.Service {
-	svcs := map[host.Name]*model.Service{}
+func findWorkloadServices(workloads []*model.WorkloadInfo, push *model.PushContext) *waypointServices {
+	ws := &waypointServices{}
 	for _, wl := range workloads {
 		for _, ns := range push.ServiceIndex.HostnameAndNamespace {
 			svc := ns[wl.Namespace]
 			if svc == nil {
 				continue
 			}
 			if labels.Instance(svc.Attributes.LabelSelectors).Match(wl.Labels) {
-				svcs[svc.Hostname] = svc
+				if ws.services == nil {
+					ws.services = map[host.Name]*model.Service{}
+				}
+				ws.services[svc.Hostname] = svc
 			}
 		}
 	}
-	return svcs
+	keys := maps.Keys(ws.services)
+	slices.Sort(keys)
```

Comment:
nit: sort by name doesn't seem right, keep consistent with other places by creation timestamp

Reply:
Thanks, updated. Just realized that the logic for sorting the workload was missing as well, which resulted in an undefined order of the custom IP match list. I've added a similar sorting logic to the workload.